[oneTBB] Add task_scheduler_handle (#358)
oneapi-src/oneAPI-spec: source/elements/oneTBB/source/deprecated/task_arena_attach_tag.rst

> .. SPDX-FileCopyrightText: 2019-2020 Intel Corporation

Reviewer: ```suggestion
.. SPDX-FileCopyrightText: 2019-2021 Intel Corporation
```

..
.. SPDX-License-Identifier: CC-BY-4.0

==================
task_arena::attach 
==================
**[deprecated.task_arena_attach_tag]**

    .. caution::

        Deprecated in oneTBB Specification 1.1.

> A set of methods for constructing a ``task_arena`` with ``attach``.

Reviewer 2: ```suggestion
A set of methods for constructing attached ``task_arena``.
```

Consider this wording. If you like it, then consider also applying it in other places.

Reviewer 2: We have decided to leave wording as is.

Reviewer 3: Discussed offline, decided to keep the wording from the old spec.

.. code:: cpp

>     // Defined in header <tbb/task_arena.h>
> 
>     namespace tbb {

Reviewer 2: ```suggestion
    // Defined in header <oneapi/tbb/task_arena.h>
    namespace oneapi {
        namespace tbb {
```
Do not forget to add a closure bracket and fix indentation

        class task_arena {
        public:
            // ...
            struct attach {};

            explicit task_arena(task_arena::attach);
            void initialize(task_arena::attach);
            // ...
        };

    } // namespace tbb

    
Member types and constants
--------------------------

.. cpp:struct:: attach

    A tag for constructing a task_arena with attach.

Member functions
----------------

.. cpp:function:: explicit task_arena(task_arena::attach)

    Creates an instance of ``task_arena`` that is connected to the internal task arena representation currently used by the calling thread.
    If no such arena exists yet, creates a ``task_arena`` with default parameters.

    .. note::

        Unlike other constructors, this one automatically initializes
        the new ``task_arena`` when connecting to an already existing arena.

        
.. cpp:function:: void initialize(task_arena::attach)

>     If an internal task arena representation currently used by the calling thread, the method ignores arena

Reviewer: Can we use task arena instead of ``task_arena``?
